<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.302">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Naz Salehin">
<meta name="dcterms.date" content="2025-09-10">

<title>simons_et_al - 08 - Niakan human</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.6/require.min.js" integrity="sha512-c3Nl8+7g4LMSTdrm621y7kf9v3SDPnhxLNhcjFJbKECVnmZHTdo+IRO05sNLTH/D3vA6u1X32ehoLC7WFVdheg==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>
<script type="application/javascript">define('jquery', [],function() {return window.jQuery;})</script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../notebooks/01_human_loadAndIntegrate.html">Reports</a></li><li class="breadcrumb-item"><a href="../notebooks/01_human_loadAndIntegrate.html">08 - Niakan human</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">simons_et_al</a> 
        <div class="sidebar-tools-main">
    <a href="https://twitter.com/LabBrickman" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/brickmanlab" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-github"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../README.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">simon_et_al</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Reports</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_human_loadAndIntegrate.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">08 - Niakan human</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/01_mouse_loadAndIntegrate.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Niakan mouse treatment (follow-up on 06 analysis)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_human_integrationPlots.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Niakan - Plots for cell type predictions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/04_mouse_integrationPlots.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Niakan - Plots for cell type predictions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../notebooks/05_human_volcanoPlotsAndUmaps.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 Human VolcanoPlotsAndUmaps</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#load-in-libraries" id="toc-load-in-libraries" class="nav-link active" data-scroll-target="#load-in-libraries"><span class="header-section-number">1</span> Load in libraries</a></li>
  <li><a href="#load-in-data" id="toc-load-in-data" class="nav-link" data-scroll-target="#load-in-data"><span class="header-section-number">2</span> Load in data</a>
  <ul class="collapse">
  <li><a href="#initial-sequencing-from-the-crick" id="toc-initial-sequencing-from-the-crick" class="nav-link" data-scroll-target="#initial-sequencing-from-the-crick"><span class="header-section-number">2.1</span> Initial sequencing from the Crick</a></li>
  <li><a href="#sequencing-from-cambridge" id="toc-sequencing-from-cambridge" class="nav-link" data-scroll-target="#sequencing-from-cambridge"><span class="header-section-number">2.2</span> Sequencing from Cambridge</a></li>
  <li><a href="#concatenate-anndata" id="toc-concatenate-anndata" class="nav-link" data-scroll-target="#concatenate-anndata"><span class="header-section-number">2.3</span> Concatenate anndata</a></li>
  </ul></li>
  <li><a href="#model-predictions" id="toc-model-predictions" class="nav-link" data-scroll-target="#model-predictions"><span class="header-section-number">3</span> Model predictions</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">08 - Niakan human</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Naz Salehin </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">September 10, 2025</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="load-in-libraries" class="level2" data-number="1">
<h2 data-number="1" data-anchor-id="load-in-libraries"><span class="header-section-number">1</span> Load in libraries</h2>
<div class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>matplotlib inline</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>load_ext autoreload</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="op">%</span>autoreload <span class="dv">2</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scvi</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> scanpy <span class="im">as</span> sc</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> pandas <span class="im">as</span> pd</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> numpy <span class="im">as</span> np</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> seaborn <span class="im">as</span> sns</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> matplotlib.pyplot <span class="im">as</span> plt</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> anndata <span class="im">as</span> ad</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="im">import</span> sys<span class="op">;</span> sys.path.append(<span class="st">"../scripts/"</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="im">from</span> helpers <span class="im">import</span> normalize_smartseq</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>GENE_LEN <span class="op">=</span> <span class="st">'/home/gkb340/Brickman/shared/references/homo_sapiens/ensembl/GRCh38_110/Homo_sapiens.GRCh38.110.gene_length.tsv'</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>REMAKE_FIGURES <span class="op">=</span> <span class="va">False</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-in-data" class="level2" data-number="2">
<h2 data-number="2" data-anchor-id="load-in-data"><span class="header-section-number">2</span> Load in data</h2>
<section id="initial-sequencing-from-the-crick" class="level3" data-number="2.1">
<h3 data-number="2.1" data-anchor-id="initial-sequencing-from-the-crick"><span class="header-section-number">2.1</span> Initial sequencing from the Crick</h3>
<div class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>crick_adata <span class="op">=</span> sc.read_h5ad(<span class="st">"../data/external/niakan_et_al/niakan_combined_matrix.h5ad"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-error">
<pre><code>NameError: name 'sc' is not defined</code></pre>
</div>
</div>
<div class="cell" data-execution_count="15">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>crick_adata.obs[<span class="st">'sample_fine'</span>] <span class="op">=</span> crick_adata.obs[<span class="st">'sample'</span>].<span class="bu">str</span>.split(<span class="st">'_'</span>, expand<span class="op">=</span><span class="va">True</span>)[<span class="dv">0</span>].to_list()</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>Simon_metadata <span class="op">=</span> pd.read_table(<span class="st">"../data/processed/Samples_LIMSID.csv"</span> ,sep <span class="op">=</span> <span class="st">','</span>)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>crick_adata.obs <span class="op">=</span> crick_adata.obs.join(Simon_metadata[[<span class="st">'LIMS.ID'</span>,<span class="st">'Embryo'</span>]].set_index(<span class="st">'LIMS.ID'</span>), on<span class="op">=</span><span class="st">'sample_fine'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="16">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>crick_adata.obs_names <span class="op">=</span> crick_adata.obs[<span class="st">'Embryo'</span>] <span class="op">+</span> <span class="st">'_'</span> <span class="op">+</span> crick_adata.obs[<span class="st">'sample'</span>]</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>crick_adata.obs[<span class="st">'treatment'</span>] <span class="op">=</span> <span class="st">'DMSO'</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>crick_adata.obs.loc[crick_adata.obs_names.<span class="bu">str</span>.contains(<span class="st">'Ulix'</span>), <span class="st">'treatment'</span>] <span class="op">=</span> <span class="st">'Ulix'</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>crick_adata.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"NIAKAN_1"</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>crick_adata.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Simon et al, 2024"</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>crick_adata.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>crick_adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="16">
<pre><code>AnnData object with n_obs × n_vars = 132 × 62754
    obs: 'sample', 'fastq_1', 'fastq_2', 'sample_fine', 'Embryo', 'treatment', 'batch', 'experiment', 'technology'
    var: 'gene_symbol'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="20">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>crick_adata.var[<span class="st">"mt"</span>] <span class="op">=</span> crick_adata.var.gene_symbol.<span class="bu">str</span>.startswith(<span class="st">"MT-"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>sc.pp.calculate_qc_metrics(crick_adata, qc_vars<span class="op">=</span>[<span class="st">"mt"</span>], inplace<span class="op">=</span><span class="va">True</span>, log1p<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>sc.pl.violin(crick_adata, [<span class="st">"n_genes_by_counts"</span>, <span class="st">"total_counts"</span>, <span class="st">"pct_counts_mt"</span>], jitter<span class="op">=</span><span class="fl">0.4</span>, multi_panel<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>sc.pl.scatter(crick_adata, <span class="st">"total_counts"</span>, <span class="st">"n_genes_by_counts"</span>, color<span class="op">=</span><span class="st">"pct_counts_mt"</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>sns.histplot(x<span class="op">=</span>crick_adata.obs[<span class="st">'pct_counts_mt'</span>], bins<span class="op">=</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-7-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-7-output-2.png"></p>
</div>
<div class="cell-output cell-output-display" data-execution_count="20">
<pre><code>&lt;Axes: xlabel='pct_counts_mt', ylabel='Count'&gt;</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-7-output-4.png"></p>
</div>
</div>
<div class="cell" data-execution_count="21">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>crick_adata <span class="op">=</span> crick_adata[crick_adata.obs.pct_counts_mt <span class="op">&lt;</span> <span class="dv">30</span>].copy()</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>crick_adata <span class="op">=</span> crick_adata[crick_adata.obs.n_genes_by_counts <span class="op">&gt;</span> <span class="dv">8000</span>].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="22">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>sc.pl.violin(crick_adata, [<span class="st">"n_genes_by_counts"</span>, <span class="st">"total_counts"</span>, <span class="st">"pct_counts_mt"</span>], jitter<span class="op">=</span><span class="fl">0.4</span>, multi_panel<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>sc.pl.scatter(crick_adata, <span class="st">"total_counts"</span>, <span class="st">"n_genes_by_counts"</span>, color<span class="op">=</span><span class="st">"pct_counts_mt"</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>sns.histplot(x<span class="op">=</span>crick_adata.obs[<span class="st">'pct_counts_mt'</span>], bins<span class="op">=</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-9-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-9-output-2.png"></p>
</div>
<div class="cell-output cell-output-display" data-execution_count="22">
<pre><code>&lt;Axes: xlabel='pct_counts_mt', ylabel='Count'&gt;</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-9-output-4.png"></p>
</div>
</div>
</section>
<section id="sequencing-from-cambridge" class="level3" data-number="2.2">
<h3 data-number="2.2" data-anchor-id="sequencing-from-cambridge"><span class="header-section-number">2.2</span> Sequencing from Cambridge</h3>
<div class="cell" data-execution_count="17">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>babraham_adata <span class="op">=</span> sc.read_h5ad(<span class="st">"../data/assays/SCR_NS_20240521/processed/GRCh38_110/combined_matrix.merged.h5ad"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="18">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>babraham_adata.obs_names <span class="op">=</span> babraham_adata.obs[<span class="st">'sample'</span>]</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>babraham_adata.obs[<span class="st">'treatment'</span>] <span class="op">=</span> <span class="st">'DMSO'</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>babraham_adata.obs.loc[babraham_adata.obs_names.<span class="bu">str</span>.contains(<span class="st">'Ulix'</span>), <span class="st">'treatment'</span>] <span class="op">=</span> <span class="st">'Ulix'</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>babraham_adata.obs[<span class="st">'batch'</span>] <span class="op">=</span> <span class="st">"NIAKAN_2"</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>babraham_adata.obs[<span class="st">'experiment'</span>] <span class="op">=</span> <span class="st">"Simon et al, 2024"</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>babraham_adata.obs[<span class="st">'technology'</span>] <span class="op">=</span> <span class="st">"SMART-seq2"</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>babraham_adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="18">
<pre><code>AnnData object with n_obs × n_vars = 100 × 62754
    obs: 'sample', 'fastq_1', 'fastq_2', 'treatment', 'batch', 'experiment', 'technology'
    var: 'gene_symbol'</code></pre>
</div>
</div>
<div class="cell" data-execution_count="23">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>babraham_adata.var[<span class="st">"mt"</span>] <span class="op">=</span> babraham_adata.var.gene_symbol.<span class="bu">str</span>.startswith(<span class="st">"MT-"</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>sc.pp.calculate_qc_metrics(babraham_adata, qc_vars<span class="op">=</span>[<span class="st">"mt"</span>], inplace<span class="op">=</span><span class="va">True</span>, log1p<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>sc.pl.violin(babraham_adata, [<span class="st">"n_genes_by_counts"</span>, <span class="st">"total_counts"</span>, <span class="st">"pct_counts_mt"</span>], jitter<span class="op">=</span><span class="fl">0.4</span>, multi_panel<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>sc.pl.scatter(babraham_adata, <span class="st">"total_counts"</span>, <span class="st">"n_genes_by_counts"</span>, color<span class="op">=</span><span class="st">"pct_counts_mt"</span>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>sns.histplot(x<span class="op">=</span>babraham_adata.obs[<span class="st">'pct_counts_mt'</span>], bins<span class="op">=</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-12-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-12-output-2.png"></p>
</div>
<div class="cell-output cell-output-display" data-execution_count="23">
<pre><code>&lt;Axes: xlabel='pct_counts_mt', ylabel='Count'&gt;</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-12-output-4.png"></p>
</div>
</div>
<div class="cell" data-execution_count="24">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>babraham_adata <span class="op">=</span> babraham_adata[babraham_adata.obs.pct_counts_mt <span class="op">&lt;</span> <span class="dv">30</span>].copy()</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>babraham_adata <span class="op">=</span> babraham_adata[babraham_adata.obs.n_genes_by_counts <span class="op">&gt;</span> <span class="dv">8000</span>].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="25">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>sc.pl.violin(babraham_adata, [<span class="st">"n_genes_by_counts"</span>, <span class="st">"total_counts"</span>, <span class="st">"pct_counts_mt"</span>], jitter<span class="op">=</span><span class="fl">0.4</span>, multi_panel<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>sc.pl.scatter(babraham_adata, <span class="st">"total_counts"</span>, <span class="st">"n_genes_by_counts"</span>, color<span class="op">=</span><span class="st">"pct_counts_mt"</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>sns.histplot(x<span class="op">=</span>babraham_adata.obs[<span class="st">'pct_counts_mt'</span>], bins<span class="op">=</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-14-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-14-output-2.png"></p>
</div>
<div class="cell-output cell-output-display" data-execution_count="25">
<pre><code>&lt;Axes: xlabel='pct_counts_mt', ylabel='Count'&gt;</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-14-output-4.png"></p>
</div>
</div>
</section>
<section id="concatenate-anndata" class="level3" data-number="2.3">
<h3 data-number="2.3" data-anchor-id="concatenate-anndata"><span class="header-section-number">2.3</span> Concatenate anndata</h3>
<div class="cell" data-execution_count="26">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>adata <span class="op">=</span> ad.concat([crick_adata, babraham_adata])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="27">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>adata.obs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="27">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th">sample</th>
<th data-quarto-table-cell-role="th">fastq_1</th>
<th data-quarto-table-cell-role="th">fastq_2</th>
<th data-quarto-table-cell-role="th">treatment</th>
<th data-quarto-table-cell-role="th">batch</th>
<th data-quarto-table-cell-role="th">experiment</th>
<th data-quarto-table-cell-role="th">technology</th>
<th data-quarto-table-cell-role="th">n_genes_by_counts</th>
<th data-quarto-table-cell-role="th">log1p_n_genes_by_counts</th>
<th data-quarto-table-cell-role="th">total_counts</th>
<th data-quarto-table-cell-role="th">log1p_total_counts</th>
<th data-quarto-table-cell-role="th">pct_counts_in_top_50_genes</th>
<th data-quarto-table-cell-role="th">pct_counts_in_top_100_genes</th>
<th data-quarto-table-cell-role="th">pct_counts_in_top_200_genes</th>
<th data-quarto-table-cell-role="th">pct_counts_in_top_500_genes</th>
<th data-quarto-table-cell-role="th">total_counts_mt</th>
<th data-quarto-table-cell-role="th">log1p_total_counts_mt</th>
<th data-quarto-table-cell-role="th">pct_counts_mt</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">Human_Ulix_5_SIM5110A133</td>
<td>SIM5110A133</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>Ulix</td>
<td>NIAKAN_1</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>12540</td>
<td>9.436759</td>
<td>16318175.0</td>
<td>16.607790</td>
<td>27.067255</td>
<td>33.021707</td>
<td>40.767923</td>
<td>53.448765</td>
<td>2596380.0</td>
<td>14.769629</td>
<td>15.910971</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Human_Ulix_4_SIM5110A88</td>
<td>SIM5110A88</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>Ulix</td>
<td>NIAKAN_1</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>13011</td>
<td>9.473627</td>
<td>16349510.0</td>
<td>16.609709</td>
<td>23.037987</td>
<td>29.586251</td>
<td>37.693747</td>
<td>51.772891</td>
<td>1259369.0</td>
<td>14.046123</td>
<td>7.702793</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Human_Ulix_5_SIM5110A127</td>
<td>SIM5110A127</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>Ulix</td>
<td>NIAKAN_1</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>12053</td>
<td>9.397152</td>
<td>6229982.0</td>
<td>15.644884</td>
<td>27.717287</td>
<td>34.285813</td>
<td>42.493606</td>
<td>56.090708</td>
<td>1005475.0</td>
<td>13.820971</td>
<td>16.139294</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Human_Ulix_4_SIM5110A85</td>
<td>SIM5110A85</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>Ulix</td>
<td>NIAKAN_1</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>14246</td>
<td>9.564302</td>
<td>24619864.0</td>
<td>17.019064</td>
<td>19.988749</td>
<td>25.531935</td>
<td>33.104822</td>
<td>46.332910</td>
<td>2569475.0</td>
<td>14.759212</td>
<td>10.436593</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Human_Ulix_5_SIM5110A142</td>
<td>SIM5110A142</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>/scratch/Brickman/pipelines/Niakan2024_human/r...</td>
<td>Ulix</td>
<td>NIAKAN_1</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>10535</td>
<td>9.262553</td>
<td>13425119.0</td>
<td>16.412638</td>
<td>23.108585</td>
<td>31.311454</td>
<td>41.168916</td>
<td>56.296395</td>
<td>788035.0</td>
<td>13.577299</td>
<td>5.869855</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
<td>...</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Human_Ulix_7_1</td>
<td>Human_Ulix_7_1</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>Ulix</td>
<td>NIAKAN_2</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>12338</td>
<td>9.420520</td>
<td>7991007.0</td>
<td>15.893827</td>
<td>20.325098</td>
<td>26.809174</td>
<td>35.067495</td>
<td>49.236786</td>
<td>538162.0</td>
<td>13.195917</td>
<td>6.734595</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Human_DMSO_6_18</td>
<td>Human_DMSO_6_18</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>DMSO</td>
<td>NIAKAN_2</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>9591</td>
<td>9.168685</td>
<td>1832853.0</td>
<td>14.421385</td>
<td>23.418299</td>
<td>29.630854</td>
<td>37.395907</td>
<td>51.251028</td>
<td>220061.0</td>
<td>12.301664</td>
<td>12.006473</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Human_Ulix_6_6</td>
<td>Human_Ulix_6_6</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>Ulix</td>
<td>NIAKAN_2</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>11746</td>
<td>9.371353</td>
<td>12517080.0</td>
<td>16.342606</td>
<td>20.012575</td>
<td>26.891527</td>
<td>35.473273</td>
<td>50.137876</td>
<td>913879.0</td>
<td>13.725454</td>
<td>7.301055</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Human_Ulix_6_30</td>
<td>Human_Ulix_6_30</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>Ulix</td>
<td>NIAKAN_2</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>13966</td>
<td>9.544453</td>
<td>27490648.0</td>
<td>17.129356</td>
<td>18.133076</td>
<td>24.556694</td>
<td>33.008141</td>
<td>47.091940</td>
<td>1456117.0</td>
<td>14.191284</td>
<td>5.296772</td>
</tr>
<tr class="odd">
<td data-quarto-table-cell-role="th">Human_Ulix_7_8</td>
<td>Human_Ulix_7_8</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>/maps/projects/dan1/data/Brickman/assays/SCR_N...</td>
<td>Ulix</td>
<td>NIAKAN_2</td>
<td>Simon et al, 2024</td>
<td>SMART-seq2</td>
<td>12450</td>
<td>9.429556</td>
<td>47063196.0</td>
<td>17.667002</td>
<td>20.350137</td>
<td>27.642863</td>
<td>36.563244</td>
<td>51.822529</td>
<td>2275928.0</td>
<td>14.637898</td>
<td>4.835897</td>
</tr>
</tbody>
</table>

<p>89 rows × 18 columns</p>
</div>
</div>
</div>
<div class="cell" data-execution_count="28">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>sc.pl.violin(adata, [<span class="st">"n_genes_by_counts"</span>, <span class="st">"total_counts"</span>, <span class="st">"pct_counts_mt"</span>], jitter<span class="op">=</span><span class="fl">0.4</span>, multi_panel<span class="op">=</span><span class="va">True</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>sc.pl.scatter(adata, <span class="st">"total_counts"</span>, <span class="st">"n_genes_by_counts"</span>, color<span class="op">=</span><span class="st">"pct_counts_mt"</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>sns.histplot(x<span class="op">=</span>adata.obs[<span class="st">'pct_counts_mt'</span>], bins<span class="op">=</span><span class="dv">30</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-17-output-1.png"></p>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-17-output-2.png"></p>
</div>
<div class="cell-output cell-output-display" data-execution_count="28">
<pre><code>&lt;Axes: xlabel='pct_counts_mt', ylabel='Count'&gt;</code></pre>
</div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-17-output-4.png"></p>
</div>
</div>
<div class="cell" data-execution_count="29">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>adata <span class="op">=</span> normalize_smartseq(adata, GENE_LEN)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>adata.layers[<span class="st">"counts"</span>] <span class="op">=</span> adata.X.copy()</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>sc.pp.normalize_total(adata)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>sc.pp.log1p(adata)</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>adata.raw <span class="op">=</span> adata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>SMART-SEQ: Normalization
SMART-SEQ: Common genes 62663</code></pre>
</div>
</div>
<div class="cell" data-execution_count="30">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>sc.pp.highly_variable_genes(adata, flavor<span class="op">=</span><span class="st">"seurat_v3"</span>, n_top_genes<span class="op">=</span><span class="dv">3_000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.1.5/lib/python3.10/site-packages/scanpy/preprocessing/_highly_variable_genes.py:75: UserWarning: `flavor='seurat_v3'` expects raw count data, but non-integers were found.
  warnings.warn(</code></pre>
</div>
</div>
<div class="cell" data-execution_count="31">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>sc.tl.pca(adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="33">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>sc.pl.pca(adata, color<span class="op">=</span>[<span class="st">'treatment'</span>, <span class="st">'batch'</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-21-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="34">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="kw">del</span> adata.varm[<span class="st">'PCs'</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="35">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>adata.write(<span class="st">"../results/niakan_08.adata.h5ad"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="model-predictions" class="level2" data-number="3">
<h2 data-number="3" data-anchor-id="model-predictions"><span class="header-section-number">3</span> Model predictions</h2>
<div class="cell" data-execution_count="36">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>lvae <span class="op">=</span> scvi.model.SCANVI.load(<span class="st">'/home/gkb340/Brickman/projects/proks-salehin-et-al-v2/results/100_human_integration/scanvi_ns_15/'</span>)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>scvi.model.SCANVI.prepare_query_anndata(adata, lvae)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>lvae_q <span class="op">=</span> scvi.model.SCANVI.load_query_data(adata, lvae)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>INFO     File                                                                                                      
         /home/gkb340/Brickman/projects/proks-salehin-et-al-v2/results/100_human_integration/scanvi_ns_15/model.pt 
         already downloaded                                                                                        
INFO     Found 100.0% reference vars in query data.                                                                </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>/projects/dan1/data/Brickman/conda/envs/scvi-1.1.5/lib/python3.10/site-packages/scvi/model/base/_utils.py:66: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  model = torch.load(model_path, map_location=map_location)
/projects/dan1/data/Brickman/conda/envs/scvi-1.1.5/lib/python3.10/site-packages/scvi/data/_manager.py:215: UserWarning: Missing labels key ct. Filling in with unlabeled category Unknown.
  field_registry[_constants._STATE_REGISTRY_KEY] = field.transfer_field(</code></pre>
</div>
</div>
<div class="cell" data-execution_count="37">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>lvae_q.train(max_epochs<span class="op">=</span><span class="dv">20</span>, plan_kwargs<span class="op">=</span><span class="bu">dict</span>(weight_decay<span class="op">=</span><span class="fl">0.0</span>), check_val_every_n_epoch<span class="op">=</span><span class="dv">10</span>, early_stopping<span class="op">=</span><span class="va">True</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>INFO     Training for 20 epochs.                                                                                   
Epoch 20/20: 100%|██████████| 20/20 [00:01&lt;00:00, 11.64it/s, v_num=1, train_loss_step=5.64e+3, train_loss_epoch=5.64e+3]Epoch 20/20: 100%|██████████| 20/20 [00:01&lt;00:00, 13.04it/s, v_num=1, train_loss_step=5.64e+3, train_loss_epoch=5.64e+3]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>GPU available: False, used: False
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
SLURM auto-requeueing enabled. Setting signal handlers.
/projects/dan1/data/Brickman/conda/envs/scvi-1.1.5/lib/python3.10/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:441: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.
/projects/dan1/data/Brickman/conda/envs/scvi-1.1.5/lib/python3.10/site-packages/lightning/pytorch/loops/fit_loop.py:293: The number of training batches (1) is smaller than the logging interval Trainer(log_every_n_steps=10). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
/projects/dan1/data/Brickman/conda/envs/scvi-1.1.5/lib/python3.10/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:441: The 'val_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=3` in the `DataLoader` to improve performance.
`Trainer.fit` stopped: `max_epochs=20` reached.</code></pre>
</div>
</div>
<div class="cell" data-execution_count="38">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>adata.obsm[<span class="st">"X_scANVI"</span>] <span class="op">=</span> lvae_q.get_latent_representation()</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>adata.obs[<span class="st">'prediction'</span>] <span class="op">=</span> lvae_q.predict()</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>adata.obs[<span class="st">'entropy'</span>] <span class="op">=</span> <span class="dv">1</span> <span class="op">-</span> lvae_q.predict(soft<span class="op">=</span><span class="va">True</span>).<span class="bu">max</span>(axis<span class="op">=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-execution_count="39">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>sc.pp.neighbors(adata)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>sc.tl.umap(adata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>2024-10-21 11:49:29.974023: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT</code></pre>
</div>
</div>
<div class="cell" data-execution_count="42">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>sc.pl.umap(adata, color<span class="op">=</span>[<span class="st">'prediction'</span>, <span class="st">'treatment'</span>,<span class="st">'batch'</span>], size <span class="op">=</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display">
<p><img src="01_human_loadAndIntegrate_files/figure-html/cell-28-output-1.png"></p>
</div>
</div>
<div class="cell" data-execution_count="43">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>sc.metrics.confusion_matrix(<span class="st">'treatment'</span>, <span class="st">'prediction'</span>, normalize<span class="op">=</span><span class="va">False</span>, data<span class="op">=</span>adata.obs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="43">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">prediction</th>
<th data-quarto-table-cell-role="th">Epiblast_6.0</th>
<th data-quarto-table-cell-role="th">Epiblast_7.0</th>
<th data-quarto-table-cell-role="th">Inner Cell Mass</th>
<th data-quarto-table-cell-role="th">Late epiblast</th>
<th data-quarto-table-cell-role="th">Primitive Endoderm</th>
<th data-quarto-table-cell-role="th">Trophectoderm_6.0</th>
<th data-quarto-table-cell-role="th">Trophectoderm_7.0</th>
<th data-quarto-table-cell-role="th">Trophectoderm_8.0</th>
<th data-quarto-table-cell-role="th">Trophectoderm_9.0</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">treatment</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">DMSO</td>
<td>1</td>
<td>3</td>
<td>1</td>
<td>1</td>
<td>7</td>
<td>3</td>
<td>9</td>
<td>1</td>
<td>10</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">Ulix</td>
<td>7</td>
<td>9</td>
<td>1</td>
<td>0</td>
<td>4</td>
<td>5</td>
<td>19</td>
<td>1</td>
<td>7</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell" data-execution_count="46">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>sc.metrics.confusion_matrix(<span class="st">'batch'</span>, <span class="st">'prediction'</span>, normalize<span class="op">=</span><span class="va">False</span>, data<span class="op">=</span>adata.obs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="46">
<div>


<table class="dataframe table table-sm table-striped small" data-quarto-postprocess="true" data-border="1">
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th">prediction</th>
<th data-quarto-table-cell-role="th">Epiblast_6.0</th>
<th data-quarto-table-cell-role="th">Epiblast_7.0</th>
<th data-quarto-table-cell-role="th">Inner Cell Mass</th>
<th data-quarto-table-cell-role="th">Late epiblast</th>
<th data-quarto-table-cell-role="th">Primitive Endoderm</th>
<th data-quarto-table-cell-role="th">Trophectoderm_6.0</th>
<th data-quarto-table-cell-role="th">Trophectoderm_7.0</th>
<th data-quarto-table-cell-role="th">Trophectoderm_8.0</th>
<th data-quarto-table-cell-role="th">Trophectoderm_9.0</th>
</tr>
<tr class="odd">
<th data-quarto-table-cell-role="th">batch</th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
<th data-quarto-table-cell-role="th"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td data-quarto-table-cell-role="th">NIAKAN_1</td>
<td>2</td>
<td>3</td>
<td>0</td>
<td>0</td>
<td>0</td>
<td>3</td>
<td>9</td>
<td>0</td>
<td>8</td>
</tr>
<tr class="even">
<td data-quarto-table-cell-role="th">NIAKAN_2</td>
<td>6</td>
<td>9</td>
<td>2</td>
<td>1</td>
<td>11</td>
<td>5</td>
<td>19</td>
<td>2</td>
<td>9</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<div class="cell" data-execution_count="44">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>adata.obs[<span class="st">'treatment'</span>].value_counts()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-display" data-execution_count="44">
<pre><code>Ulix    53
DMSO    36
Name: treatment, dtype: int64</code></pre>
</div>
</div>
<div class="cell" data-execution_count="45">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>adata.write(<span class="st">"../results/niakan_08.withPredictions.adata.h5ad"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
});
</script>
</div> <!-- /content -->



</body></html>